packages/esm-patient-common-lib/src/visits/useInfiniteVisits.ts

```diff
@@ -0,0 +1,53 @@
+import { openmrsFetch, restBaseUrl, useConfig } from '@openmrs/esm-framework';
+import useSWRInfinite from 'swr/infinite';
+
+export function useInfiniteVisits(patientUuid: string) {
+  const getKey = useInfiniteVisitsGetKey()(patientUuid);
+
+  const {
+    data,
+    error,
+    isLoading,
+    isValidating,
+    mutate: localMutate,
+    size,
+    setSize,
+  } = useSWRInfinite(patientUuid ? getKey : null, openmrsFetch, { parallel: true });
+
+  return {
+    visits: data ? [].concat(data?.flatMap((page) => page?.data?.results)) : null,
+    error,
+    hasMore: data?.length ? !!data[data.length - 1].data?.links?.some((link) => link.rel === 'next') : false,
+    isLoading,
+    isValidating,
+    mutateVisits: localMutate,
+    setSize,
+    size,
+  };
+}
+
+export function useInfiniteVisitsGetKey() {
+  const { numberOfVisitsToLoad } = useConfig({ externalModuleName: '@openmrs/esm-patient-chart-app' });
+  const customRepresentation =
+    'custom:(uuid,location,encounters:(uuid,diagnoses:(uuid,display,rank,diagnosis,voided),form:(uuid,display),encounterDatetime,orders:full,obs:(uuid,concept:(uuid,display,conceptClass:(uuid,display)),display,groupMembers:(uuid,concept:(uuid,display),value:(uuid,display),display),value,obsDatetime),encounterType:(uuid,display,viewPrivilege,editPrivilege),encounterProviders:(uuid,display,encounterRole:(uuid,display),provider:(uuid,person:(uuid,display)))),visitType:(uuid,name,display),startDatetime,stopDatetime,patient,attributes:(attributeType:ref,display,uuid,value)';
+
+  return (patientUuid: string) => {
```

Comment:
Is there a reason not to make `patientUuid` just a parameter to this?

Reply:
I mean I see what you did with `useMutateVisit()` I just don't see it being used at a point where we can't also pass in the patientUuid as a parameter there.